libs/application-generic/src/services/resource-validator.service.ts

```diff
@@ -0,0 +1,86 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import {
+  EnvironmentRepository,
+  NotificationTemplateRepository,
+} from '@novu/dal';
+import { FeatureFlagsKeysEnum } from '@novu/shared';
+
+import { GetFeatureFlag, NotificationStep } from '../usecases';
+
+@Injectable()
+export class ResourceValidatorService {
+  private readonly MAX_STEPS_PER_WORKFLOW = 10;
+  private readonly MAX_WORKFLOWS_LIMIT = 100;
+
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private environmentRepository: EnvironmentRepository,
+    private getFeatureFlag: GetFeatureFlag,
+  ) {}
+
+  async validateStepsLimit(environmentId: string, steps: NotificationStep[]) {
+    const isWorkflowLimitEnabled = await this.getFeatureFlag.execute({
+      key: FeatureFlagsKeysEnum.IS_MAX_STEPS_PER_WORKFLOW_ENABLED,
+      environmentId,
+      organizationId: 'system',
+      userId: 'system',
+    });
+
+    if (!isWorkflowLimitEnabled) {
+      return;
+    }
+
+    if (steps.length > this.MAX_STEPS_PER_WORKFLOW) {
+      throw new BadRequestException({
+        message: `Workflow steps limit exceeded. Maximum allowed steps is ${this.MAX_STEPS_PER_WORKFLOW}, but got ${steps.length} steps.`,
+        providedStepsCount: steps.length,
+        maxSteps: this.MAX_STEPS_PER_WORKFLOW,
+      });
+    }
+  }
+
+  async validateWorkflowLimit(environmentId: string) {
+    const environment = await this.getEnvironment(environmentId);
+
+    const workflowsCount = await this.notificationTemplateRepository.count({
+      _environmentId: environmentId,
+    });
+
+    if (workflowsCount < this.MAX_WORKFLOWS_LIMIT) {
+      return;
+    }
+
+    const maxWorkflowLimit = await this.getFeatureFlag.getNumber({
+      key: FeatureFlagsKeysEnum.MAX_WORKFLOW_LIMIT_NUMBER,
+      contextKey: 'environment',
```

Comment:
This DX is a bit cumbersome. In every request for a feature flag you need to have at least one of the following:

```
- organizationId,
- userId,
- environmentId,
```

This is the DX we had so far for boolean. So for numeric we should aim for a very similar experience.
```
await this.getFeatureFlag.execute(
      GetFeatureFlagCommand.create({
        key: FeatureFlagsKeysEnum.IS_API_IDEMPOTENCY_ENABLED,
        environmentId,
        organizationId,
        userId,
      })
    );
```

That is, execute needs to get a generic that is either boolean or number and the result will be of type generic.

<img width="1187" alt="Screenshot 2025-02-13 at 12 28 25" src="https://github.com/user-attachments/assets/a581d7c0-fa2f-48fd-aadf-655de21286f0" />

Reply:
i would go even further and not use the use case, and will use the service directly. 
if we want to simplify the flow this is the way to do it, this usecase is redundant at the moment.
please be aware that the launchdackly already have a generic return type as you mentioned